app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,41 @@
+class VotesController < ApplicationController
+  def upvote
+    found = Votes.find_by(user: params[:user], api: params[:api_id])
+
+    if found == nil
+      Votes.create(
+        user: User.find(params[:user]),
+        api: Api.find(params[:api_id]),
+        user_vote: true)
+      render json: Votes.calculate_votes(params[:api_id])
+      return
+    end
+    if found.user_vote == false
```

Comment:
Any particular reason you're opening another `if` instead of using an `elsif` branch of your conditional? I'd recommend the latter unless I'm missing something about this choice!